### PR TITLE
Implemented adding an offer to draft

### PR DESCRIPTION
**Pull request by Yurenko**
Implemented adding an offer to the draft.
Also when you are the author of this offer you can change the status of this offer on offer details page.
Fixed list of offers on the find offer page and carousel. It should be with active status.

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/83120263/bbd99b56-1043-4bff-90b0-f6d729cdda98

### CODE REVIEW

File: src/containers/offer-page/create-offer/CreateOffer.tsx [hunk 1]
```diff
@@ -132,8 +132,9 @@ const CreateOffer: FC<CreateOfferProps> = ({ closeDrawer }) => {
           {t(`offerPage.createOffer.buttonTitles.${userRole}`)}
         </AppButton>
         <AppButton
-          disabled
+          onClick={() => handleNonInputValueChange('status', 'draft')}
```
Comment (tsivkadi) [inline src/containers/offer-page/create-offer/CreateOffer.tsx:135]: use StatusEnum.draft

Comment (MickhaelSh) [reply src/containers/offer-page/create-offer/CreateOffer.tsx:135]: and move it outside

File: src/pages/offer-details/OfferDetails.tsx [hunk 2]
```diff
@@ -48,54 +48,93 @@ const OfferDetails = () => {
...
+        ? StatusEnum.Active
+        : StatusEnum.Draft
+
+    if (offerData) {
+      void fetchData(newStatus)
+      void fetchDataOffer()
+    }
+  }
 
   const buttonActions = [
```
Comment (MickhaelSh) [inline src/pages/offer-details/OfferDetails.tsx:92]: move this to constants as a func

File: src/pages/offer-details/OfferDetails.tsx [hunk 3]
```diff
@@ -48,54 +48,93 @@ const OfferDetails = () => {
...
     defaultResponse: null,
     onResponseError: responseError
   })
 
+  const updateOffer = useCallback(
+    (status?: StatusEnum) => OfferService.updateOffer(id, { status }),
+    [id]
+  )
+
+  const { fetchData } = useAxios<Offer | null, StatusEnum>({
```
Comment (MickhaelSh) [inline src/pages/offer-details/OfferDetails.tsx:66]: rename fetchdata as you did higher

**Review by MickhaelSh — APPROVED** [review]

File: src/containers/offer-page/create-offer/CreateOffer.tsx (hunk 1 above)
Comment (Yurenko) [reply src/containers/offer-page/create-offer/CreateOffer.tsx:135]: Fixed

File: src/pages/offer-details/OfferDetails.tsx (hunk 3 above)
Comment (Yurenko) [reply src/pages/offer-details/OfferDetails.tsx:66]: Fixed

File: src/pages/offer-details/OfferDetails.tsx (hunk 2 above)
Comment (Yurenko) [reply src/pages/offer-details/OfferDetails.tsx:92]: Fixed

**Review by tsivkadi — APPROVED** [review]

**Review by FryDay969 — APPROVED** [review]

**Review by mxrcury — APPROVED** [review]

**Review by ArturBekhDEV — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**